### PR TITLE
[docs] Recipe for a single wire for a bundle in a module interface

### PR DESCRIPTION
**Pull request by oharboe**
### Contributor Checklist

- [x] Did you add Scaladoc to every public function/method?
- [x] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [x] Did you add appropriate documentation in `docs/src`?
- [x] Did you state the API impact?
- [x] Did you specify the code generation impact?
- [x] Did you request a desired merge strategy?
- [x] Did you add text to be included in the Release Notes for this change?

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels?
- [ ] Did you mark the proper milestone (Bug fix: `3.3.x`, [small] API extension: `3.4.x`, API modification or big change: `3.5.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you mark as `Please Merge`?

### CODE REVIEW

File: docs/src/cookbooks/cookbook.md [hunk 1]
```diff
@@ -404,6 +406,39 @@ class ModuleWithOptionalIO(flag: Boolean) extends Module {
 }
 ```
 
+### How do I create a single input/output wire for a bundle?
+
+If you have a bundle with many elements in a module interface, the interface can become
+quite verbose. To simplify the module interface, the entire bundle
+can be represented with a single wire.
+
```
Comment (jackkoenig) [inline docs/src/cookbooks/cookbook.md:414]: This recipe is very similar to "How do I create a UInt from an instance of a Bundle?" and "How do I create a Bundle from a UInt?", but I totally agree that having the right framing of recipes is important for discoverability. 

What do you think about phrasing it as something like "How do I pack an Aggregate into a single input/output port?" and then pointing it to "How do I create a UInt from an instance of a Bundle?"? We could also tweak the example up there to provide a port as well (instead of just a wire).

File: docs/src/cookbooks/cookbook.md [hunk 2]
```diff
@@ -404,6 +406,39 @@ class ModuleWithOptionalIO(flag: Boolean) extends Module {
...
+Consider:
+
+```scala mdoc:silent:reset
+import chisel3._
+
+class DooHickeyBundle extends Bundle {
```
Comment (mwachs5) [inline docs/src/cookbooks/cookbook.md:420]: this is a bit nitpicky, but `DooHickey` is like `Foo` which we use most places, is there a reason to not use that here? I think `DooHickey` might be confusing for people who aren't native (american?) english speakers, while Foo is practically a term of art at this point...

File: docs/src/cookbooks/cookbook.md [hunk 3]
```diff
@@ -404,6 +406,39 @@ class ModuleWithOptionalIO(flag: Boolean) extends Module {
...
+  // Lots of individual wires
+  val a = Vec(5, Bool())
+}
+
+class DooHickey() extends Module {
+  // many wires
+  val b = IO(Input(new DooHickeyBundle))
+  // single module interface wire wire
```
Comment (mwachs5) [inline docs/src/cookbooks/cookbook.md:428]:
```suggestion
  // one wide wire
```

File: docs/src/cookbooks/cookbook.md (hunk 1 above)
Comment (oharboe) [reply docs/src/cookbooks/cookbook.md:414]: "Aggregate" is not used elsewhere in the cookbook, but rather "Bundle" is.

Tweaked exampel as suggested.

**Comment by oharboe** [comment]
@mwachs5 @jackkoenig Anything else?

File: docs/src/cookbooks/cookbook.md [hunk 4]
```diff
@@ -62,26 +64,35 @@ class Foo extends RawModule {
 
 Use the [`asTypeOf`](https://www.chisel-lang.org/api/latest/chisel3/UInt.html#asTypeOf[T%3C:chisel3.Data](that:T):T) method to reinterpret the [`UInt`](https://www.chisel-lang.org/api/latest/chisel3/UInt.html) as the type of the [`Bundle`](https://www.chisel-lang.org/api/latest/chisel3/Bundle.html).
 
+If you have a bundle with many elements in a module interface,
```
Comment (mwachs5) [inline docs/src/cookbooks/cookbook.md:67]: I am not sure what you are trying to do in this PR. You replaced an example that went with the text above, `"Use the .asTyepeOf"` with a different example that is showing your different point. Why did you delete the example that was there before?

**Review by mwachs5 — CHANGES_REQUESTED** [review]
You are deleting an example that went with the text before and interjecting a different example / use case. Shouldn't this go under a different heading? I am also not sure why this is explicitly "* [How do I create a single input/output wire for a Bundle?]" and that is not even the title of the section that you are linking to.

I suggest you leave the existing examples alone and add a new top-level "recipe" heading to show whatever it is you are trying to show.
